src/crimson/osd/osd_operations/snaptrim_event.cc

```diff
@@ -195,7 +196,7 @@ SnapTrimEvent::start()
   }, [this](std::exception_ptr eptr) -> snap_trim_ertr::future<seastar::stop_iteration> {
     logger().debug("{}: interrupted {}", *this, eptr);
     return crimson::ct_error::eagain::make();
-  }, pg).finally([this] {
+  }, pg).finally([this, ref] {
```

Comment:
Does the issue disappear after this fix?

See https://github.com/ceph/ceph/blob/main/src/crimson/osd/shard_services.h#L164-L169

IIUC there is no difference with this change because the life of op is already extended by its caller above.

Reply:
Ah, sorry about refering to an irrelavent logic.

Please refer to https://github.com/ceph/ceph/blob/main/src/crimson/osd/shard_services.h#L142-L147 instead, which should be the caller of `SnapTrimObjEvent::start()` and has already extended its life to `finally()` IIUC.